ibm-ups/main.cpp

```diff
@@ -42,6 +42,9 @@ int main(int argc, const char* argv[])
         auto event = sdeventplus::Event::get_default();
         bus.attach_event(event.get(), SD_EVENT_PRIORITY_NORMAL);
 
+        // Obtain D-Bus service name
+        bus.request_name("xyz.openbmc_project.Power.IBMUPS");
```

Comment:
If you move this after creating  the Monitor object then you don't need to emit any interfaces added signal. Usually that's the proper way to do it.

Reply:
Matt and I discussed this offline.  Andrew's state manager code needs the InterfaceAdded signals.